### PR TITLE
Issue #6367: Improve pitest coverage for RightCurly

### LINKED ISSUE #6367: Fix pitest coverage for RightCurly
TODO:
revert commit - https://github.com/checkstyle/checkstyle/commit/3316f0b1a179c3ecd2ca3c21d9a81129282ba630 and https://github.com/checkstyle/checkstyle/commit/63d9db3900848b3d662383aefb5c336978f549a0

fix pitest coverage to be 100%, and kill all mutations, RightCurly should have no survived mutations.

discussion of coverage issues: https://github.com/checkstyle/checkstyle/pull/4247#pullrequestreview-189430650

-------

<a href="https://www.bountysource.com/issues/68536186-fix-pitest-coverage-for-rightcurly">

</a>

### PR DESCRIPTION
**Pull request by fzdy1914**
address one survival of #6367

### CODE REVIEW

**Comment by fzdy1914** [comment]
I may need help with this pr.
When I try to write the test, I wrote the following block:
```
    { 
        {  int a; }    
    }
```
The outside block is identified as INSTANCE_INIT, but the inner block cannot be identified as INSTANCE_INIT. Can I know the reason why this happens?

**Comment by rnveach** [comment]
> The outside block is identified as INSTANCE_INIT, but the inner block cannot be identified as INSTANCE_INIT. 

Because it isn't an instance initializer. `INSTANCE_INIT` is the block of code that is executed before a constructor is called. It is similar to the static initializer except it is connected to the class instance. You can't have a instance initializer inside an instance initializer. The second set of braces is just an ordinary `SLIST`.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java [hunk 1]
```diff
@@ -27,6 +27,13 @@ void method2(java.util.HashSet<String> set) {
             put("polygene", "lubricants");
             put("alpha", "betical");
         }}; //NO violation
+
+        java.util.Map<String, String> map3 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+        }{};
+        }; // violation
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java]: please move comment to line 35 , where there is a violation,

**Comment by fzdy1914** [comment]
I am still investigating if I can fix other mutations

**Comment by romani** [comment]
It is better to merge this, no need to do all as one commit, we always contiously improve.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java (hunk 1 above)
Comment (fzdy1914) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java]: Fixed

**Comment by fzdy1914** [comment]
@romani Ready to review

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java [hunk 2]
```diff
@@ -27,6 +27,13 @@ void method2(java.util.HashSet<String> set) {
             put("polygene", "lubricants");
             put("alpha", "betical");
         }}; //NO violation
+
+        java.util.Map<String, String> map3 = new java.util.LinkedHashMap<String, String>() {{
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java]: Please put this test case near the bottom of the file so line numbers in test aren't changed.

Comment (fzdy1914) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAlone.java]: changed as request

**Comment by rnveach** [comment]
@romani The changes you requested were done, and we will split this into multiple PRs.
I will merge this after the release once the CI is green.

**Comment by romani** [comment]
Thanks a lot for for making commit more clear.

**Pull request merged**